### PR TITLE
Added Blockchain Document Loader

### PR DESCRIPTION
**Pull request by saginawj**
This PR contains a Document Loader and Integration Test for Blockchain which:

- Uses the Alchemy API which provides read access to the major blockchain projects
- Currently only Ethereum Mainnet is used
- Currently the load function returns a collection of NFTs at the provided contractAddress.  

Future areas of extension:

- Extend to additional blockchains (e.g. Polygon) and environments (e.g. testnets)
- Extend to additional APIs beyond NFTs (e.g. Blockchain Transactions)

An ALCHEMY_API_KEY is required to access the Alchemy API.  This can be set as an environment variable. 

I'm currently using this loader locally to analyze NFT attributes in a Q&A pipeline.  I figured this could be useful for others.

Please provide any feedback!

### CODE REVIEW

File: langchain/document_loaders/blockchain.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import os
+import requests
+import json
+
+from typing import List
+
+from langchain.docstore.document import Document
+from langchain.document_loaders.base import BaseLoader
+
+ALCHEMY_API_KEY = os.environ.get("ALCHEMY_API_KEY", "docs-demo")
```
Comment (hwchase17) [inline langchain/document_loaders/blockchain.py:10]: lets not do this here, but rather inside init

**Review by hwchase17 — COMMENTED** [review]
lets an example notebook! otherwise looks pretty good

File: langchain/document_loaders/blockchain.py (hunk 1 above)
Comment (saginawj) [reply langchain/document_loaders/blockchain.py]: Made the update in latest commit- thanks!

**Comment by saginawj** [comment]
> lets an example notebook! otherwise looks pretty good

Here's a [sample notebook](https://colab.research.google.com/drive/15qQA24fEREoTgzdWKALOcAxXA4ePNsys?usp=sharing) on my repo.  In this example, I ask about popular attributes for NFTs in the specified contract address (Bored Apes Yacht Club in this example).

A free api key from [Alchemy ](https://www.alchemy.com/) is required to run it. Sometimes the demo version works. 

image

**Pull request merged**